### PR TITLE
New vector store specification, and pinecone namespace feature

### PR DESCRIPTION
**Pull request by adrian-lorenzo**
# Changes

1. **Changed `VectorStore` specification.** Now, the `search` method expects an `itemsLimit` as a parameter instead of a fixed number of 10 items. The return type of the `search` method now returns a type that includes four essential values for most of the use cases: `id` of the item, `content`, `metadata` and the `score` obtained from the match.
2. **Added configurable Pinecone namespace** to allow users to select which Namespace of their index they want to use.
3. Fixed typo in `learn` method input in `SemanticSearch`: Now, the type is called `LearnInput`

### CODE REVIEW

**Review by juanjoman — APPROVED** [review]
LGTM!

File: modules/semanticsearch/src/main/kotlin/com/theagilemonkeys/ellmental/semanticsearch/SemanticSearch.kt [hunk 1]
```diff
@@ -35,18 +36,19 @@ class SemanticSearch {
      * a ranked list of results.
      *
      * @param text The text to be used as reference for semantic search.
+     * @param itemsLimit The maximum number of results to return.
      * @return A list of semantically similar texts ranked by semantic distance (the closest first).
      */
-    suspend fun search(text: String): SearchOutput {
+    suspend fun search(text: String, itemsLimit: Int): SearchOutput {
```
Comment (BorjaZarco) [inline modules/semanticsearch/src/main/kotlin/com/theagilemonkeys/ellmental/semanticsearch/SemanticSearch.kt:42]:
```suggestion
    suspend fun search(text: String, itemsLimit?: Int): SearchOutput {
```

Can we make the itemsLimit optional? Perhaps also adding a default of 5 or 10

**Review by doramasma — APPROVED** [review]
LGMT!

**Review by BorjaZarco — APPROVED** [review]
LGTM! 👍

**Review by javiertoledo — COMMENTED** [review]
Great improvements 🎉

**Pull request merged**